docker/compute_worker/compute_worker.py

```diff
@@ -151,7 +152,40 @@ def __init__(self, run_args):
         submission_api_url_parsed = urlparse(self.submissions_api_url)
         websocket_host = submission_api_url_parsed.netloc
         websocket_scheme = 'ws' if submission_api_url_parsed.scheme == 'http' else 'wss'
-        self.websocket_url = f"{websocket_scheme}://{websocket_host}/"
+        self.websocket_url = f"{websocket_scheme}://{websocket_host}/submission_input/{self.user_pk}/{self.submission_id}/{self.secret}/"
+
+    async def watch_file(self):
+        if not self.detailed_results_url:
+            return
+        file_path = os.path.join(self.output_dir, 'detailed_results.html')
+        last_modified_time = None
+        while self.watch:
+            if os.path.exists(file_path):
+                new_time = os.path.getmtime(file_path)
+                if new_time != last_modified_time:
+                    last_modified_time = new_time
+                    await self.send_file(file_path)
+                await asyncio.sleep(5)
+            else:
+                await asyncio.sleep(5)
```

Comment:
could do...

```python

            if os.path.exists(file_path):
                new_time = os.path.getmtime(file_path)
                if new_time != last_modified_time:
                    last_modified_time = new_time
                    await self.send_file(file_path)

            await asyncio.sleep(5)
```

(since you sleep in both sides of the conditional)